flatfs.go

```diff
@@ -682,7 +700,18 @@ func (fs *Datastore) Query(q query.Query) (query.Results, error) {
 	})
 	go b.Process.CloseAfterChildren() //nolint
 
-	return b.Results(), nil
+	results := b.Results()
+
+	if len(q.Filters) > 0 ||
+		len(q.Orders) > 0 ||
+		q.Limit > 0 ||
+		q.Offset > 0 {
+		nq := q
+		nq.Prefix = "" // already handled
+		results = query.NaiveQueryApply(nq, results)
+	}
```

Comment:
So `NaiveQueryApply` will already do nothing if it does not have to do anything so you might skip the checks here as redundant: https://github.com/ipfs/go-datastore/blob/master/query/query_impl.go#L119

Reply:
Ah, well, https://github.com/ipfs/go-datastore/pull/147 (now merged).

Reply:
You are setting prefix to `""` anyways so it would have not entered the `if`